org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -1321,7 +1348,12 @@ class CodeActionContext {
  * These commands are typically code fixes to either fix problems or to beautify/refactor code.
  */
 @JsonRpcData
-class CodeActionParams {
+class CodeActionParams implements WorkDoneProgressParams, PartialResultParams {
+	
+	Either<String, Number> workDoneToken
```

Comment:
Given that this type is referenced very often, you could try to define
```xtend
class ProgressToken extends Either<String, Number> {
	override toString() {
		if (isLeft)
			left
		else if (isRight)
			right.toString
	}
}
```
and replace `Either<String, Number>` with this new class (i.e. redefine the type instead of wrapping it as in the previous version). I think parsing/serializing should still work the same with the EitherTypeAdapter.

Reply:
You would also have to create a protected constructor and static creation methods as in the superclass (but I'd call them both `create` instead of `forLeft` and `forRight`).